jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java

```diff
@@ -513,6 +515,13 @@ public void accept(SelectableChannel selectable, Map<String, Object> context)
         }
     }
 
+    public void onClientConnection(ClientConnectionListener listener)
+    {
+        if (this.listener != null)
+            return;
+        this.listener = listener;
+    }
```

Comment:
What's this?

Reply:
I would like to prevent creating multiple event listeners for the same `ClientConnector` instance. If you think it is not reasonable then I can fix it.

Reply:
@arsenalzp I would like to support multiple listeners if necessary.

This is how it should work:

```java
ClientConnector c = ...;
c.addEventListener(new ClientConnector.ConnectListener() {
   ...
});
```

Then, when you need to emit the events, you can do:

```java
getBeans(ClientConnector.ConnectListener.class).forEach(listener -> ...);
```

Reply:
You'll want a `.getListeners()` and maybe even a `.removeListener()` methods then too.

Reply:
@joakime they are already there, in the super class.